js/components/discussions/threads.vue

```diff
@@ -29,46 +27,121 @@
     },
+    events: {
+        /**
+         * Add the created discussion in the list and display it
+         * @type {Discussion} the newly created discussion
+         */
+        'discussion:created': function(discussion) {
+            this.hideForm()
+            this.discussions.unshift(discussion);
+            this.$nextTick(() => {
+                // Scroll to new discussion when displayed and expand it
+                const $thread = this.threadFor(discussion.id);
+                $thread.detailed = true;
+                this.$scrollTo($thread);
+            });
+        },
+        /**
+         * Replace the updated discussion in the list
+         * @type {Discussion} the updated discussion
+         */
+        'discussion:updated': function(discussion) {
+            const index = this.discussions.indexOf(this.discussions.find(d => d.id == discussion.id));
+            // See: https://v1.vuejs.org/guide/list.html#Array-Change-Detection
+            this.discussions.$set(index, discussion);
+        }
+    },
     ready() {
         this.$api.get('discussions', {for: this.subjectId}).then(response => {
             this.discussions = response.data;
+            if (document.location.hash) {
+                this.$nextTick(() => { // Wait for data to be binded
+                    this.jumpToHash(document.location.hash);
+                });
+            }
         }).catch(log.error.bind(log));
-
-        this.$on('discussions-load', discussions => {
-            this.discussions.unshift(discussions);
-        });
     },
     methods: {
+        /**
+         * Display the start discussion form or triggers an authentication if required
+         */
         displayForm() {
+            if (!Auth.need_user(this._('You need to be logged in to start a discussion.'))) {
+                return;
+            }
             this.formDisplayed = true;
         },
 
         /**
+         * Close the discussion form
+         */
+        hideForm() {
+            this.formDisplayed = false;
+        },
+
+        /**
          * Trigger a new prefilled discussion.
          */
         start(title, comment) {
-            this.formDisplayed = true;
+            this.displayForm()
             // Wait for next tick because the form needs to be visible to scroll
             this.$nextTick(() => {
-                this.$scrollTo(this.$els.form);
-                this.$refs.form.prefill(title, comment);
+                if (this.$els.form && this.$refs.form) { // Avoid logging errors
+                    this.$scrollTo(this.$els.form);
+                    this.$refs.form.prefill(title, comment);
+                }
             })
+        },
+
+        /**
+         * Get the thread compoent for a given discussion id
+         */
+        threadFor(id) {
+            return this.$refs.threads.find($thread => $thread.discussion.id == id);
+        },
+
+        /**
+         * Handle deep linking
+         */
+        jumpToHash(hash) {
```

Comment:
Nop, the component is loaded after the hash trigger and so the ids are not present for resolution and scroll. And for comment or new comment links, the component needs to be expanded because you can't scroll to a non visible component.